lib/ethpy/ethpy/hyperdrive/api/api_test.py

```diff
@@ -23,102 +25,208 @@
+        assert pool_info_to_fixedpoint(pool_info) == hyperdrive_interface.current_pool_state.pool_info
 
     def test_checkpoint(self, local_hyperdrive_pool: DeployedHyperdrivePool):
         """Checks that the Hyperdrive checkpoint matches what is returned from the smart contract.
 
         All arguments are fixtures.
         """
-        uri: URI | None = cast(HTTPProvider, local_hyperdrive_pool.web3.provider).endpoint_uri
-        rpc_uri = uri if uri else URI("http://localhost:8545")
-        abi_dir = "./packages/hyperdrive/src/abis"
-        hyperdrive_contract_addresses: HyperdriveAddresses = local_hyperdrive_pool.hyperdrive_contract_addresses
-        eth_config = EthConfig(artifacts_uri="not used", rpc_uri=rpc_uri, abi_dir=abi_dir)
-        hyperdrive = HyperdriveInterface(eth_config, addresses=hyperdrive_contract_addresses)
-        checkpoint_id = hyperdrive.calc_checkpoint_id(block_timestamp=hyperdrive.current_pool_state.block_time)
+        hyperdrive_interface = self.setup_hyperdrive_interface(local_hyperdrive_pool)
+        checkpoint_id = hyperdrive_interface.calc_checkpoint_id(
+            block_timestamp=hyperdrive_interface.current_pool_state.block_time
+        )
         # TODO: remove cast when pypechain consolidates dataclasses.
-        checkpoint = cast(Checkpoint, hyperdrive.hyperdrive_contract.functions.getCheckpoint(checkpoint_id).call())
-        print(f"{checkpoint=}")
-        assert checkpoint_to_fixedpoint(checkpoint) == hyperdrive.current_pool_state.checkpoint
+        checkpoint = cast(
+            Checkpoint, hyperdrive_interface.hyperdrive_contract.functions.getCheckpoint(checkpoint_id).call()
+        )
+        assert checkpoint_to_fixedpoint(checkpoint) == hyperdrive_interface.current_pool_state.checkpoint
 
     def test_spot_price_and_fixed_rate(self, local_hyperdrive_pool: DeployedHyperdrivePool):
-        """Checks that the Hyperdrive spot price and fixed rate match computing it by hand.
-
-        All arguments are fixtures.
-        """
-        uri: URI | None = cast(HTTPProvider, local_hyperdrive_pool.web3.provider).endpoint_uri
-        rpc_uri = uri if uri else URI("http://localhost:8545")
-        hyperdrive_contract_addresses: HyperdriveAddresses = local_hyperdrive_pool.hyperdrive_contract_addresses
-        hyperdrive = HyperdriveInterface(
-            eth_config=EthConfig(
-                artifacts_uri="not used",
-                rpc_uri=rpc_uri,
-                abi_dir="./packages/hyperdrive/src/abis",
-            ),
-            addresses=hyperdrive_contract_addresses,
-        )
+        """Checks that the Hyperdrive spot price and fixed rate match computing it by hand."""
+        hyperdrive_interface = self.setup_hyperdrive_interface(local_hyperdrive_pool)
         # get pool config variables
-        pool_config = hyperdrive.current_pool_state.pool_config
+        pool_config = hyperdrive_interface.current_pool_state.pool_config
```

Comment:
can you rename all instances to `interface`

Reply:
done in de95a90aef1d2b1dfb287877f562d6aaca46f7ec